pvanalytics/quality/time.py

```diff
@@ -184,3 +184,103 @@ def shifts_ruptures(event_times, reference_times, period_min=2,
     # localize the shift_amount series to the timezone of the input
     shift_amount = shift_amount.tz_localize(event_times.index.tz)
     return shift_amount != 0, shift_amount
+
+
+def _has_dst(events, date, window, min_difference):
+    """Return True if `events` appears to have a daylight savings shift
+    on `date`.
+
+    Parameters
+    ----------
+    events : Series
+        Series of timestamps, one per day.
+    date : Timestamp
+        Timestamp corresponding to midnight on the day to check for daylight
+        savings shifts.
+    window : int
+        Number of days before and after `date` to use when checking for
+        for shifts in `events`
+    min_different : int
+        Minimum difference between mean event time before `date` and mean
+        event time after `date` for a shift tom be detected.
+
+    Returns
+    -------
+    bool
+        Whether or not a shift of at least `min_difference` minutes was
+        detected in the the event times on `date`.
+    """
+    before = events[date - window:date - pd.Timedelta(days=1)]
+    after = events[date:date + window]
+    if len(before) == 0 or len(after) == 0:
+        raise ValueError(f"Insufficient data at {date}.")
+    before = before.dt.hour * 60 + before.dt.minute
+    after = after.dt.hour * 60 + after.dt.minute
+    return abs(before.mean() - after.mean()) > min_difference
+
+
+def has_dst(events, tz, window=7, min_difference=45):
+    """Return true if `events` appears to have daylight-savings shifts
+    at the dates in `shift_dates`.
+
+    Compares the mean event time in minutes since midnight over the
+    `window` days before and after each date in `shift_dates`. If the
+    difference is greater than `min_difference` then a shift has occurred
+    on that date.
+
+    Parameters
+    ----------
+    events : Series
+        Series with one timestamp for each day. The timestamp should
+        correspond to an event that occurs at roughly the same time on
+        each day, and shifts with daylight savings transitions. For example,
+        you may pass sunrise, sunset, or solar transit time.
+    tz : str
+        Name of a timezone that observes daylight savings and has the same
+        or similar UTC offset as the expected time zone for `events`.
+    window : int, default 7
+        Number of days before and after the shift date to consider. When
+        passing rounded timestamps in `events` it may be necessary to
+        use a smaller window. [days]
+    min_difference : int, default 45
+        Minimum difference between the mean event time before the shift
+        date and the mean event time after the event time. If the difference
+        is greater than `min_difference` a shift has occurred on that date.
+        [minutes]
+
+    Returns
+    -------
+    Series
+        Boolean Series indexed by dates on which Daylight Savings transitions
+        are expected to occur in `tz` and True if a shift was found on that
+        date in `events`
```

Comment:
Return series with same index as `events`. All False except the transition dates which are True if a dst-like transition was observed across that date.